packages/agreement-process/src/services/agreementActivationProcessor.ts

```diff
@@ -177,23 +178,46 @@ async function activateAgreement(
     ...updatedAgreementSeed,
   };
 
-  const updateAgreementEvent = toCreateEventAgreementUpdated(
-    updatedAgreement,
-    agreementData.metadata.version,
-    correlationId
-  );
+  const activationEvent = await match(firstActivation)
+    .with(true, async () => {
+      const activatedAgreementEvent = toCreateEventAgreementActivated(
+        updatedAgreement,
+        agreementData.metadata.version,
+        correlationId
+      );
 
-  if (firstActivation) {
-    await createContract(
-      updatedAgreement,
-      updatedAgreementSeed,
-      eservice,
-      consumer,
-      attributeQuery,
-      tenantQuery,
-      storeFile
-    );
-  }
+      await createContract(
+        updatedAgreement,
+        updatedAgreementSeed,
+        eservice,
+        consumer,
+        attributeQuery,
+        tenantQuery,
+        storeFile
+      );
+
+      return activatedAgreementEvent;
```

Comment:
the activation event should now contain also the reference to the newly created contract